ledger-core/v2/insolar/contract/virtual.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package contract
+
+import (
+	"github.com/insolar/assured-ledger/ledger-core/v2/reference"
+)
+
+// MethodFunc is a typedef for wrapper contract header
+type MethodFunc func(oldState []byte, args []byte) (newState []byte, result []byte, err error)
+
+func ConstructorIsolation() MethodIsolation {
+	return MethodIsolation{
+		Interference: CallTolerable,
+		State:        CallDirty,
+	}
+}
+
+type MethodIsolation struct {
+	Interference InterferenceFlag
+	State        StateFlag
+}
+
+// Method is a struct for Method and it's properties
+type Method struct {
+	Func      MethodFunc
+	Isolation MethodIsolation
+}
+
+// Methods maps name to contract method
+type Methods map[string]Method
+
+// Constructor is a typedef of typical contract constructor
+type Constructor func(ref reference.Global, args []byte) (state []byte, result []byte, err error)
+
+// Constructors maps name to contract constructor
+type Constructors map[string]Constructor
+
+// Wrapper stores all needed about contract wrapper (it's methods/constructors)
+type Wrapper struct {
+	GetCode      MethodFunc
+	GetPrototype MethodFunc
+
+	Methods      Methods
+	Constructors Constructors
+}
+
+//go:generate stringer -type=PendingState
+
+// PendingState is a state of execution for each object
+type PendingState int
+
+const (
+	PendingUnknown PendingState = iota // PendingUnknown signalizes that we don't know about execution state
+	NotPending                         // NotPending means that we know that this task is not executed by another VE
+	InPending                          // InPending means that we know that method on object is executed by another VE
+)
+
+func (i PendingState) Equal(other PendingState) bool {
+	return i == other
+}
+
+type SideEffectType uint8
+
+const (
+	SideEffectTypeNone SideEffectType = iota
```

Comment:
maybe we should move that declaration to separate package and remove prefix SideEffect?
If not, then rename `SideEffectTypeNone ` -> `SideEffectNone `

Reply:
As a side effect of moving to another package we can `go gen` Stringer

Reply:
I'm not sure that we even need this, both SideEffect and PendingState, probably will just delete all of them, as they are not used

Reply:
SideEffectType was used, as i remember, when we determine what should we do with the result of execution, or am i wrong?

Reply:
it is different side effect type defined in requestresult package :(